### PR TITLE
feat: Sync history page

### PR DESCRIPTION
**Pull request by kiram15**
Instead of having a modal, now the sync history goes to a new link. The url pattern is `http://localhost:1991/test-enterprise/admin/settings/lms/[LMS]/[ID]`. The horizontal card has the same functionality as the cards on the main page, but also has the two sync history tables. 

Screen Shot 2023-01-05 at 10 53 14 AM

# For all changes

- [x] Ensure adequate tests are in place (or reviewed existing tests cover changes)

# Only if submitting a visual change

- [x] Ensure to attach screenshots
- [x] Ensure to have UX team confirm screenshots

### CODE REVIEW

File: src/data/services/LmsApiService.js [hunk 1]
```diff
@@ -127,16 +127,24 @@ class LmsApiService {
     return LmsApiService.apiClient().post(LmsApiService.providerDataSyncUrl, formData);
   }
 
-  static postNewMoodleConfig(formData) {
-    return LmsApiService.apiClient().post(`${LmsApiService.lmsIntegrationUrl}/moodle/configuration/`, formData);
+  static fetchBlackboardGlobalConfig() {
```
Comment (kiram15) [inline src/data/services/LmsApiService.js:130]: Alphabetized this list

File: src/utils.js [hunk 2]
```diff
@@ -245,45 +245,52 @@ const channelMapping = {
...
     delete: LmsApiService.deleteCanvasConfig,
   },
   [CORNERSTONE_TYPE]: {
     displayName: 'Cornerstone',
     icon: CornerstoneIcon,
+    get: LmsApiService.fetchSingleCornerstoneConfig,
     update: LmsApiService.updateCornerstoneConfig,
     delete: LmsApiService.deleteCornerstoneConfig,
   },
+  [DEGREED_TYPE]: {
```
Comment (kiram15) [inline src/utils.js:266]: Also alphabetized this

File: src/components/settings/SettingsLMSTab/utils.js [hunk 3]
```diff
@@ -18,3 +20,15 @@ export const isExistingConfig = (configs, value, existingInput) => {
   }
   return false;
 };
+
+export const getStatus = (config) => {
```
Comment (marlonkeating) [inline src/components/settings/SettingsLMSTab/utils.js:24]: Can we at least reference the 'config' object schema used here?

File: src/components/settings/SettingsLMSTab/utils.js [hunk 4]
```diff
@@ -18,3 +20,15 @@ export const isExistingConfig = (configs, value, existingInput) => {
...
+  if (!isEmpty(config.isValid[0].missing)
+      || !isEmpty(config.isValid[1].incorrect)) {
+    return 'Incomplete';
+  }
+
+  if (config.active) {
+    return 'Active';
+  }
+  return 'Inactive';
```
Comment (marlonkeating) [inline src/components/settings/SettingsLMSTab/utils.js:33]: nit
```suggestion
    return config.active ? 'Active' : 'Inactive';
```

File: src/components/settings/SettingsLMSTab/ErrorReporting/SyncHistory.jsx [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+    setReloadPage(false);
+  }, [configChannel, configId, reloadPage, vars]);
+
+  // useEffect(() => {
+  //   console.log("hello love <3");
+  //   console.log(config);
+  // }, [config]);
+
+  const getLastSync = () => {
+    if (config.lastSyncErroredAt != null) {
```
[the remark this answers is not in the document]
Comment (kiram15) [reply src/components/settings/SettingsLMSTab/ErrorReporting/SyncHistory.jsx:53]: It shows the timestamp of the last error that happened. In this case, IMO we want to prioritize showing the recent sync error that happened two days ago even if a good one happened yesterday.

File: src/components/settings/SettingsLMSTab/ErrorReporting/ErrorReportingTable.jsx [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  Tab, Tabs,
+} from '@edx/paragon';
+import ContentMetadataTable from './ContentMetadataTable';
+import LearnerMetadataTable from './LearnerMetadataTable';
+
+const ErrorReportingTable = ({ config }) => {
+  const [key, setKey] = useState('contentMetadata');
+  // notification for tab must be a non-empty string to appear
+  const contentError = config.lastContentSyncErroredAt == null ? null : ' ';
+  const learnerError = config.lastLearnerSyncErroredAt == null ? null : ' ';
```
[the remark this answers is not in the document]
Comment (kiram15) [reply src/components/settings/SettingsLMSTab/ErrorReporting/ErrorReportingTable.jsx:13]: I don't see this logic used in other places, this is just to determine if we want to have a notification for the tab. @marlonkeating could you clarify?

File: src/components/settings/SettingsLMSTab/ExistingLMSCardDeck.jsx [hunk 7]
```diff
@@ -15,52 +14,29 @@ const ExistingLMSCardDeck = ({
...
       onClick={onClick}
       openError={openError}
-      openReport={openReport}
-      setReportConfig={setReportConfig}
       setErrorModalText={setErrorModalText}
       getStatus={getStatus}
     />
   ));
-  const listInactive = configData.filter(config => getStatus(config) !== 'active').map(config => (
+  const listInactive = configData.filter(config => getStatus(config) !== 'Active').map(config => (
```
Comment (marlonkeating) [inline src/components/settings/SettingsLMSTab/ExistingLMSCardDeck.jsx:32]: Import 'Active' string from ExistingCard.jsx?

File: src/components/settings/SettingsLMSTab/ErrorReporting/utils.jsx [hunk 8]
```diff
@@ -43,6 +43,7 @@ export function getSyncStatus(status, statusMessage) {
               <Popover.Content>
                 <h5 className="mb-2"><Error className="text-danger-500 mr-1" />Error<br /></h5>
                 {statusMessage || 'Something went wrong.  Please contact enterprise customer support.'}
+
```
Comment (marlonkeating) [inline src/components/settings/SettingsLMSTab/ErrorReporting/utils.jsx:46]: nit: extra line

**Review by alex-sheehan-edx — APPROVED** [review]
Couple of non-blocking questions/comments. LGTM!

File: src/components/settings/SettingsLMSTab/ErrorReporting/SyncHistory.jsx [hunk 9]
```diff
@@ -0,0 +1,238 @@
...
+
+import {
+  ACTIVATE_TOAST_MESSAGE, BLACKBOARD_TYPE, CANVAS_TYPE, CORNERSTONE_TYPE, DEGREED_TYPE,
+  DEGREED2_TYPE, errorToggleModalText, INACTIVATE_TOAST_MESSAGE, MOODLE_TYPE, SAP_TYPE,
+} from '../../data/constants';
+
+import { channelMapping } from '../../../../utils';
+import ErrorReportingTable from './ErrorReportingTable';
+
+const SyncHistory = ({ enterpriseId }) => {
```
Comment (marlonkeating) [inline src/components/settings/SettingsLMSTab/ErrorReporting/SyncHistory.jsx:24]: enterpriseId doesn't appear to be used anywhere in the component

File: src/setupTest.js [hunk 10]
```diff
@@ -6,7 +6,6 @@ import Adapter from 'enzyme-adapter-react-16';
 import MockAdapter from 'axios-mock-adapter';
 import ResizeObserverPolyfill from 'resize-observer-polyfill';
 import { getAuthenticatedHttpClient } from '@edx/frontend-platform/auth';
-import 'jest-canvas-mock';
```
[the remark this answers is not in the document]
Comment (kiram15) [reply src/setupTest.js:9]: idk but lint told me it was unused and to remove it!

**Pull request merged**